libglusterfs/src/statedump.c

```diff
@@ -300,28 +300,51 @@ gf_proc_dump_xlator_mem_info_only_in_use(xlator_t *xl)
     return;
 }
 
-/* Currently this dumps only mallinfo. More can be built on here */
+/* Currently this dumps only mallinfo. More can be built on here. */
+
+#if defined(HAVE_MALLINFO2)
+
+#define GF_MALLINFO()                                                          \
+    struct mallinfo2 info;                                                     \
+    memset(&info, 0, sizeof(info));                                            \
+    info = mallinfo2()
+
+#define GF_DICT_MINFO_VALUE(dict, option, value)                               \
+    dict_set_uint64((dict), (option), (value))
+
+#define M_FMT "%zd"
+
+#elif defined(HAVE_MALLINFO)
+
+#define GF_MALLINFO()                                                          \
+    struct mallinfo info;                                                      \
+    memset(&info, 0, sizeof(info));                                            \
+    info = mallinfo()
+
+#define GF_DICT_MINFO_VALUE(dict, option, value)                               \
+    dict_set_int32((dict), (option), (value))
```

Comment:
I think we could always use uint64, even if we are using `struct mallinfo`.

Reply:
OK